pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex_test.go

```diff
@@ -1307,6 +1307,8 @@ func TestWorkloadsForWaypoint(t *testing.T) {
 	// Add a namespace waypoint to the pod
 	s.addWaypoint(t, "10.0.0.1", "waypoint-ns", constants.WorkloadTraffic, true)
 	s.addWaypoint(t, "10.0.0.2", "waypoint-sa1", constants.WorkloadTraffic, true)
+	// Wait until waypoints are available
+	assert.EventuallyEqual(t, func() int { return len(s.waypoints.List("")) }, 2)
```

Comment:
Should we update it in other places too where we add waypoints (e.g. right after [1220](https://github.com/istio/istio/blob/9144856a4b7a47cc64ed96b327fe8bee9e4c3452/pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex_test.go#L1220), etc.) ? Why just this one?